### ISSUE TITLE
Comment on 3.2.7 Change of Content

### ISSUE THREAD

**Issue by jnurthen** [w3c/wcag21#502]
In Single Page Web Applications a common way to communicate a change of content is to move focus to the beginning of the new content. In reading this text this would not be an allowed way of meeting this SC. 

Most times this is the method which works best for everyone (i.e. I press the search button - the results appear and focus is moved to the search results heading. A screen reader will read the heading, focus will move to the results for a screen magnifier user etc.) This needs to be an allowed method to meet this SC and according to my reading it is not currently.

**Comment by DavidMacDonald** [comment]
HI James, I'm open to trying to craft an exception along these lines. My concern is that forcing the focus to the new content might cause the user to miss some of the existing content between the control and the new content.

How could we manage that?

**Comment by jnurthen** [comment]
@DavidMacDonald I don't think you can prevent that from being a possibility but I don't see that as a reason to not allow it as a solution.

**Comment by DavidMacDonald** [comment]
I think its wading into change of context territory to pick up a screen reader user and move them down the page, skipping over unread content. 

It seems a little like the well meaning bystander who pulls the person who is blind by the arm across the street and says "let me help you", only to be told that that they were waiting for someone on the corner and didn't want to go across the street.

Am I missing something?

**Comment by jnurthen** [comment]
Yes. Why would it be useful for me to use (for example) and aria live region to provide a message to a screen reader user that search results have been returned? The most logical thing to do if someone searches for something by pressing a search button is to move the user to the results. This works for a screen reader user and works for a keyboard user.
Adding a live region when you manage focus correctly in this situation helps no one and adds bloat to the user experience and the page.

Moving focus is one of the major ways of informing a user that an action has taken place in single page web applications. There is a nice article at 
https://simplyaccessible.com/article/spangular-accessibility/#focus which talks about some of the issues.

**Comment by patrickhlauke** [comment]
If the move of focus is the result of the user actively hitting something like a search submit/refine button, then it's not an unexpected change of context. If, however, the content of the page updates dynamically as the user is interacting/typing a search query/etc and not actively hitting an action/submit button, then yes a live region is the better option as yanking focus would indeed be a change of context. We need to be careful to distinguish these two scenarios.

**Comment by mbgower** [comment]
While I certainly think a well-planned move of focus in response to a user action is a possible technique, I'm not sure that such a technique would contradict this SC as written; A change in focus is potentially covered by the admittedly vague exception "user has been advised...as a result of using control".
Further, I also think the discussion about search results points to how one could use a live region to surface info on screen (e.g., "21 results") which would be useful information in addition to the focus relocation.
In other words, the relocation may be either a sufficent or complementary technique.

**Comment by awkawk** [comment]
Thanks for your comment. The approach you describe is regularly used, and would provide a scenario where this SC would be automatically conformed to. If there are no changes of content that occur without changes of context then the web page will pass this SC.
